File: common/src/main/java/org/sandboxpowered/silica/network/login/clientbound/LoginSuccess.java
package org.sandboxpowered.silica.network.login.clientbound;

import org.sandboxpowered.silica.network.*;

import java.util.UUID;

public class LoginSuccess implements Packet {
    private UUID uuid;
    private String username;

    public LoginSuccess(UUID uuid, String username) {
        this.uuid = uuid;
        this.username = username;
    }

    public LoginSuccess() {
    }

    @Override
    public void read(PacketByteBuf buf) {
        uuid = buf.readUUID();
        username = buf.readString(16);
    }

    @Override
    public void write(PacketByteBuf buf) {
        buf.writeUUID(uuid);
        buf.writeString(username);
    }

    @Override
    public void handle(PacketHandler packetHandler, Connection connection) {

    }
}

File: common/src/main/kotlin/org/sandboxpowered/silica/network/login/clientbound/LoginSuccess.kt
package org.sandboxpowered.silica.network.login.clientbound

import org.sandboxpowered.silica.network.Connection
import org.sandboxpowered.silica.network.Packet
import org.sandboxpowered.silica.network.PacketByteBuf
import org.sandboxpowered.silica.network.PacketHandler
import java.util.*

class LoginSuccess(private var uuid: UUID?, private var username: String?) : Packet {

    constructor() : this(null, null)

    override fun read(buf: PacketByteBuf) {
        uuid = buf.readUUID()
        username = buf.readString(16)
    }

    override fun write(buf: PacketByteBuf) {
        buf.writeUUID(uuid!!)
        buf.writeString(username!!)
    }

    override fun handle(packetHandler: PacketHandler, connection: Connection) {}
}